todoman/model.py

```diff
@@ -577,9 +580,13 @@ def todos(self, all=False, lists=[], urgent=False, location='',
 
         if not all:
             # XXX: Duplicated logic of Todo.is_completed
-            extra_where.append('AND completed_at is NULL '
-                               'AND status != "CANCELLED" '
-                               'AND status != "COMPLETED"')
+            if complete:
+                extra_where.append('AND status != "CANCELLED" '
+                                   'AND status == "COMPLETED"')
```

Comment:
`status != "CANCELLED"` isn't necessary; anything matching `status == "COMPLETED` will already be not-cancelled (can't have two statuses :P ).